File: 0049-group-anagrams/0049-group-anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        res = {}
        for s in strs:
            key = tuple(sorted(s))
            if key in res:
                res[key].append(s)
            else:
                res[key] = [s]
        return list(res.values())
        

File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String,List<String>> res = new HashMap<>();
        for(String s : strs){
            char[] ab = s.toCharArray();
            Arrays.sort(ab);
            String sortstr = String.valueOf(ab);
            res.putIfAbsent(sortstr,new ArrayList<>());
            res.get(sortstr).add(s);

        }

        return new ArrayList<>(res.values());
    }
}